src/Compilers/CSharp/Portable/Lowering/IteratorRewriter/IteratorRewriter.cs

```diff
@@ -137,34 +137,28 @@ private void GenerateEnumeratorImplementation()
             {
                 var disposeMethod = OpenMethodImplementation(
                     IDisposable_Dispose,
-                    debuggerHidden: true,
-                    generateDebugInfo: false,
                     hasMethodBodyDependency: true);
 
-                var moveNextMethod = OpenMethodImplementation(
-                    IEnumerator_MoveNext,
-                    methodName: WellKnownMemberNames.MoveNextMethodName,
-                    debuggerHidden: IsDebuggerHidden(this.method),
-                    hasMethodBodyDependency: true);
+                var moveNextMethod = OpenMoveNextMethodImplementation(IEnumerator_MoveNext);
 
                 GenerateMoveNextAndDispose(moveNextMethod, disposeMethod);
             }
 
             // Add T IEnumerator<T>.Current
             {
-                OpenPropertyImplementation(IEnumeratorOfElementType_get_Current, debuggerHidden: true, hasMethodBodyDependency: false);
```

Comment:
Oh, I see, it is always "true" except in the new subclass you've created. Looks good.